oodeel/methods/vim.py

```diff
@@ -174,14 +176,16 @@ def _score_tensor(self, inputs: TensorType) -> np.ndarray:
             np.ndarray: scores
         """
         # extract features
-        features, logits = self.feature_extractor(inputs)
+        all_features, info = self.feature_extractor.predict(inputs)
+        features = all_features[0]
+        logits = info["logits"]
         features = self.op.flatten(features)
         logits = self.op.convert_to_numpy(logits)
         # vim score
         res_scores = self._compute_residual_score_tensor(features)
         energy_scores = logsumexp(logits, axis=-1)
         scores = -self.alpha * res_scores + energy_scores
-        return -np.array(scores)
+        return -np.array(scores), info
```

Comment:
I think this function should return `scores, logits` since it has no access to the labels, and then info dict would be built in `score` function of the parent class. (same remark for all detectors)